libs/template/validators_test.go

```diff
@@ -74,3 +75,56 @@ func TestValidatorInt(t *testing.T) {
 	err = validateInteger("abc")
 	assert.ErrorContains(t, err, "expected type integer, but value is \"abc\"")
 }
+
+func TestTemplateValidateType(t *testing.T) {
+	// assert validation passing
+	err := validateType(int(0), jsonschema.IntegerType)
+	assert.NoError(t, err)
+	err = validateType(int32(1), jsonschema.IntegerType)
+	assert.NoError(t, err)
+	err = validateType(int64(1), jsonschema.IntegerType)
+	assert.NoError(t, err)
+
+	err = validateType(float32(1.1), jsonschema.NumberType)
+	assert.NoError(t, err)
+	err = validateType(float64(1.2), jsonschema.NumberType)
+	assert.NoError(t, err)
+	err = validateType(int(1), jsonschema.NumberType)
```

Comment:
is int64 also number type?
also below we say int(0) is IntergerType, is it okay if it's both Integer and Number type?

Reply:
You make a good point here. For now, it's not required to permit int64 (and other int types) as a number. We can default to being more restrictive here, and if need be loosen this up. 

done, removed integers as valid numbers in validator.